the-collab-lab/tcl-25-smart-shopping-list
# Add a new item to the shopping list

## Description
This PR presents in the add view a form to the user to add a new purchasing item. After the user clicks the add button, the item is added to the database associated with the user's token and displayed in the list view.

## Related Issue

closes #5 

## Acceptance Criteria

- User is presented with a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again (Soon, Kind of Soon, Not Soon)
- When the user submits the form, the item is saved to the database, associated with the user’s token
- Along with the item name, an integer corresponding to the estimated number of days until next purchase is saved: 7 for “Soon”, 14 for “Kind of Soon”, and 30 for “Not Soon”
- The last purchased date should be set to null initially (i.e. you can create an item without purchasing it)
- Item names should be displayed as a simple, unordered list on the list view after they’re added
## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

 
## Testing Steps / QA Criteria

- From your terminal, pull down this branch with ` git pull origin ci-ia-add-new-item` and check that branch out with `git checkout ci-ia-add-new-item `
- Then `npm start to launch the app`
- Fill out the form with a new item. If you don't have a token already in your local storage, the added item will be saved in the `LA` collection of the database [here ](https://console.firebase.google.com/project/tcl-25-shopping-list/firestore/data~2FLA~2FDyGyD2ZsV66IP8OEDi09) and be displayed in the list view.
